pkg/trust/trust.go

```diff
@@ -76,15 +83,17 @@ func DefaultPolicyPath(sys *types.SystemContext) string {
 
 // RegistriesDirPath returns a path to registries.d
 func RegistriesDirPath(sys *types.SystemContext) string {
-	systemRegistriesDirPath := "/etc/containers/registries.d"
-	if sys != nil {
-		if sys.RegistriesDirPath != "" {
-			return sys.RegistriesDirPath
-		}
-		if sys.RootForImplicitAbsolutePaths != "" {
-			return filepath.Join(sys.RootForImplicitAbsolutePaths, systemRegistriesDirPath)
-		}
+	if sys != nil && sys.RegistriesDirPath != "" {
+		return sys.RegistriesDirPath
+	}
+	userRegistriesDirPath := filepath.Join(homedir.Get(), userRegistriesDir)
```

Comment:
Should there be a rootless check here?

Reply:
To keep consistent with part of the logic of c/image [newConfigWrapper ](https://github.com/containers/image/blob/f1a6f9b50d0286bf208935d4b9ab48117294ac31/pkg/sysregistriesv2/system_registries_v2.go#L351), if per-user dir exists, direclty use it, without checking whether its rootless or not

Reply:
But what happens if the directory does not exists and you are rootless.  I would figure we still want to create it.

Reply:
you mean need to create the $HOME/.config/containers/registries.conf.d if rootless? This  RegistriesDirPath only used by `podman image trust show` to see if the configuration for specific registry exist, so we don't change it if no  configuration exists. `podman image sign` will create the place for signature storage.

Reply:
Ok I thought that was what was calling this.